great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -333,6 +334,17 @@ def _save_project_config(self) -> None:
         """  # noqa: E501
         return self.variables.save()
 
+    @public_api
+    def enable_analytics(self, enable: Optional[bool]) -> None:
+        """
+        Enable or disable analytics for this DataContext.
+        With non-ephemeral contexts, this can be preserved via context.variables.save().
+
+        If set to None, the `GX_ANALYTICS_ENABLED` environment variable will be used.
+        """
+        self.config.analytics_enabled = enable
+        self._init_analytics()
```

Comment:
The rest of the variables are set with `context.variables.xyz` and `context.variables.save()` - do we care that we're deviating here?

Reply:
I originally wasn't too worried about deviating from the `context.variables.xyz = ` pattern, but this will still require a `context.variables.save()`, which is non-obvious. Maybe worth thinking about how to make it work with a property setter on `self.config.analytics_enabled`

Reply:
Could we just add the `context.variables.save()` call here (and perhaps do that for all variables)?

Reply:
Seems reasonable to me.

Reply:
Though it does differ from what we are doing with other models.